src/language/concurrency.md

```diff
@@ -16,87 +19,108 @@ prevpage:
   }
 </style>
 
-Dart supports concurrent programming with async-await, isolates, and
-classes such as `Future` and `Stream`.
-This page gives an overview of async-await, `Future`, and `Stream`,
-but it's mostly about isolates.
-
-Within an app, all Dart code runs in an _isolate._
-Each Dart isolate has a single thread of execution and
-shares no mutable objects with other isolates.
-To communicate with each other,
-isolates use message passing.
-Many Dart apps use only one isolate, the _main isolate_.
-You can create additional isolates to enable
-parallel code execution on multiple processor cores.
-
-Although Dart's isolate model is built with underlying primitives
-such as processes and threads
-that the operating system provides,
-the Dart VM's use of these primitives
-is an implementation detail that this page doesn't discuss.
+This page contains a conceptual overview of how concurrent programming works in 
+Dart. It explains the event-loop, async language features, and isolates from
+a high-level. For more practical code examples of using async features and
+isolates, read the [Asynchrony support](/language/async) and 
+[Isolates](/language/isolates) pages. 
 
-## Asynchrony types and syntax
+Concurrent programming in Dart refers to both asynchronous APIs, like `Future`
+and `Stream`, and *isolates*, which allow you to move processes to separate
+cores.
 
-If you're already familiar with `Future`, `Stream`, and async-await,
-then you can skip ahead to the [isolates section][].
+All Dart code runs in isolates, starting in the default main isolate, 
+and optionally expanding to whatever subsequent isolates you 
+explicitly create. When you spawn a new isolate, 
+it has its own isolated memory heap, and its own event loop. 
+The event loop is what makes asynchronous and 
+concurrent programming possible in Dart.
 
-[isolates section]: #how-isolates-work
+## Event Loop
 
+Dart’s runtime model is based on an event loop. 
+The event loop is responsible for executing your program's code,
+collecting and processing events, 
+running garbage collection processes, and more.
```

Comment:
@ericwindmill Lasse said garbage collection probably isnt part of the event loop [here](https://github.com/dart-lang/site-www/pull/5332#discussion_r1402421621)